pddl/requirements.py

```diff
@@ -46,6 +46,19 @@ def strips_requirements(cls) -> Set["Requirements"]:
             Requirements.CONDITIONAL_EFFECTS,
         }
 
+    @classmethod
+    def adl_requirements(cls) -> Set["Requirements"]:
+        """Get the ADL requirements."""
+        return {
+            Requirements.STRIPS,
+            Requirements.TYPING,
+            Requirements.NEG_PRECONDITION,
+            Requirements.DIS_PRECONDITION,
+            Requirements.EQUALITY,
+            Requirements.QUANTIFIED_PRECONDITION,
```

Comment:
I guess quantified preconditions covers both existential and universal, but under that path, wouldn't STRIPS cover all those here?

https://github.com/AI-Planning/pddl/blob/f54e1878109fed88ca56f3a18f3ecc7dd9fbe9eb/pddl/requirements.py#L42

Actually, I'm a bit confused as to why those are all included in STRIPS...

Reply:
Yes, quantified preconditions covers both universal and existential.
 
Thanks for pointing out the issue. I think it all depends on what reference we are following and what we want to achieve. At this moment, STRIPS covers some of the other requirements but shouldn't (based on the original version and the BNF). I don't remember why we put such additional requirements under STRIPS, but most probably to parse some domains that are not fully compliant with the original grammar... ?

Reply:
I'd probably flag it as an issue for the PDDL, and keep the `pddl` lib pure and true to the BNF. Want to tackle the slight rewrite of the STRIPS set in this PR, or open a new one?

Reply:
Agree! I can do that here..